dwst/scripts/particles.js

```diff
@@ -91,16 +106,22 @@ function skipExpressionClose(parsee) {
 
 function skipArgListOpen(parsee) {
   const argListOpen = '(';
-  parsee.read(argListOpen);
+  if (parsee.read(argListOpen) === false) {
+    const msg = `missing ${argListOpen}`;
+    throw new InvalidParticles(msg);
+  }
 }
 
 function skipArgListClose(parsee) {
   const argListClose = ')';
-  parsee.read(argListClose);
+  if (parsee.read(argListClose) === false) {
+    const msg = `missing ${argListClose}`;
```

Comment:
I think this is dead code at the moment. Not sure what to think about it. Doesn't look too dangerous.

Reply:
Maybe you could do
```
  if (parsee.read(argListClose) === false) {
    throw new Error('unexpected return value');
  }
```
Since `parsee.read(argListClose)` should always be true.

Reply:
The respective ```if``` was missing from skipArgListOpen, which caused tests to fail in InvalidParticlesExamples. I added it here too for symmetrys sake. You seem to be correct, example ```'${foo(}'``` does not make tests fail if said ```if``` is missing from skipArgListClose, though I dont understand why.

Reply:
Right. Certainly the missing ```if``` wont make the tests fail, because then the missing ')' is just ignored. Example ```'${foo(}'``` results in InvalidParticles because after ```(``` we expect arguments in readInstructionArg(parsee).

Reply:
Not only that but with our current particles implementation it is impossible to create a test that would reach the error inside this `if`.

Reply:
I'm fine leaving the code this way since it is not too complex and it seems useful if other parts of the code changes. I just wanted to point out that this code is untested at the moment and also will never be reached in the current form of the app. We could of course write a separate test for `skipArgListClose` but it is perhaps not necessary since it is not too complex and by no means critical code.

Reply:
```if (parsee.read(argListClose) === false) {
    throw new Error('unexpected return value');
  }```
I like this more, I feel it is safer if some strange changes happen in the future.